content/blog/tidymodels-2024-q1/index.Rmd

```diff
@@ -0,0 +1,203 @@
+
+```{r}
+#| echo: FALSE
+#| results: asis
+
+cat(tm_data$news, sep = "\n")
+```
+
+We'll highlight a few especially notable changes below: new prediction options in censored, consistency in augmenting parsnip models and workflows, as well as a new autoplot type for workflow sets.
+
+```{r}
+#| message: FALSE
+library(tidymodels)
+library(censored)
+```
+
+
+## New prediction options in censored
+
+As part of the framework-wide integration of survival analysis, also censored, the parsnip extension package for censored regression, received some love in the form of new prediction options.
+
+Random forests with the  `"aorsf"` engine can now predict survival time, thanks to the new feature in the [aorsf](https://docs.ropensci.org/aorsf/) package itself. This means that all engines in censored can now predict survival time.
+
+Let's predict survival time for the first five rows of the lung cancer dataset, survival analysis' `mtcars`. 
+
+```{r aorsf}
+rf_spec <- rand_forest() |>
+  set_engine("aorsf") |>
+  set_mode("censored regression")
+
+rf_fit <- rf_spec |>
+  fit(Surv(time, status) ~ age + sex, data = lung)
+
+lung_5 <- lung[1:5, ]
+predict(rf_fit, new_data = lung_5, type = "time")
+```
+
+Some models allow for predictions based on different values for tuning parameter without having to refit the model. In parsnip, we refer to this as ["the submodel trick."](https://parsnip.tidymodels.org/articles/Submodels.html) Some of those models are regularized models fitted with the [glmnet](https://glmnet.stanford.edu/) engine. In censored, the corresponding `multi_predict()` method has now gained the prediction types `"time"` and `"raw"` in addition to the existing types `"survival"` and `"linear_pred"`. 
+
+Let's fit a regularized Cox model to illustrate. Note how we set the `penalty` to a fixed value of `0.1`.
+
+```{r coxnet-fit}
+cox_fit <- proportional_hazards(penalty = 0.1) |>
+  set_engine("glmnet") |>
+  set_mode("censored regression") |>
+  fit(Surv(time, status) ~ ., data = lung)
+```
+
+Predictions made with `predict()` use that penalty value of 0.1. With `multi_predict()`, we can change that value to something different without having to refit. Conveniently, we can predict for multiple penalty values as well.
+
+```{r coxnet-pred}
+predict(cox_fit, new_data = lung_5, type = "time")
+
+mpred <- multi_predict(cox_fit, new_data = lung_5, type = "time", 
+                       penalty = c(0.01, 0.1)) 
+mpred
+```
+
+The resulting tibble is nested by observation to follow the convention of one row per observation. For each observation, the predictions are stored in a tibble containing the penalty value along with the prediction.
+
+```{r coxnet-pred-detail}
+mpred$.pred[[2]]
+```
+
+You can see that the predicted value from `predict()` matches the predicted value from `multi_predict()` with a penalty of 0.1.
```

Comment:
Shouldn't this be `0.01`? Since we are showing the submodel trick

Reply:
and if it isn't, I think we should show the `0.01` prediction 😄

Reply:
We are showing both! (Are you looking only at the source or also the rendered version?)

I mean that you can see the penalty values for the multi-predict output, find the one that says penalty = 0.1 and see that the (prediction) value is the same as the one from predict() -- because that is the penalty value we specified for the fit.  

If that is very confusing right now, I'm happy about suggestions!

Reply:
sorry, I was only looking at source! then this is good!